test/value-set.js

```diff
@@ -0,0 +1,109 @@
+/**
+* Copyright (c) 2017-present, Jembi Health Systems NPC.
+* All rights reserved.
+*
+* This source code is licensed under the BSD-style license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict'
+
+const env = require('./test-env/init')()
+const server = require('../lib/server')
+const tap = require('tap')
+const request = require('request')
+
+const basicValueSetTest = (t, test) => {
+  env.initDB((err, db) => {
+    t.error(err)
+
+    server.start((err) => {
+      t.error(err)
+
+      test(db, () => {
+        env.clearDB((err) => {
+          t.error(err)
+          server.stop(() => {
+            t.end()
+          })
+        })
+      })
+    })
+  })
+}
+
+tap.test('ValueSet Resource', { autoend: true }, (t) => {
+  t.test('should return ValueSets matching the url parameter', (t) => {
+    const valueSet = {
```

Comment:
We have a few sample ValueSet resources in the "tests/resources" folder we could use instead of defining this in the test

Reply:
Having the value set defined like this makes it much easier to follow the tests from one file rather than looking in a bunch of places. Test shouldn't really share test data. It becomes confusing. I think we should rather move to doing things this way instead and rather just have util function so that add test data is easy.

Reply:
Ok cool thats fine with me, The main reason i like having it in a file and gets used by various tests, is that it ensures all tests use the same defined resource schema, instead of defining it specifically per test and potentially having mis-matched/incorrect schema definitions tested